pynws/simple_nws.py

```diff
@@ -159,47 +171,50 @@ def _unique_alert_ids(alerts):
         """Return set of unique alert_ids."""
         return {alert[ALERT_ID] for alert in alerts}
 
-    def _new_alerts(self, alerts, current_alerts):
+    def _new_alerts(self: SimpleNWS, alerts, current_alerts):
         """Return new alerts."""
         current_alert_ids = self._unique_alert_ids(current_alerts)
         return [alert for alert in alerts if alert[ALERT_ID] not in current_alert_ids]
 
-    async def update_alerts_forecast_zone(self):
+    async def update_alerts_forecast_zone(self: SimpleNWS):
         """Update alerts zone."""
         alerts = await self.get_alerts_forecast_zone()
         new_alerts = self._new_alerts(alerts, self._alerts_forecast_zone)
         self._alerts_forecast_zone = alerts
         return new_alerts
 
-    async def update_alerts_county_zone(self):
+    async def update_alerts_county_zone(self: SimpleNWS):
         """Update alerts zone."""
         alerts = await self.get_alerts_county_zone()
         new_alerts = self._new_alerts(alerts, self._alerts_county_zone)
         self._alerts_county_zone = alerts
         return new_alerts
 
-    async def update_alerts_fire_weather_zone(self):
+    async def update_alerts_fire_weather_zone(self: SimpleNWS):
         """Update alerts zone."""
         alerts = await self.get_alerts_fire_weather_zone()
         new_alerts = self._new_alerts(alerts, self._alerts_fire_weather_zone)
         self._alerts_fire_weather_zone = alerts
         return new_alerts
 
-    async def update_alerts_all_zones(self):
+    async def update_alerts_all_zones(self: SimpleNWS):
         """Update all alerts zones."""
-        if not self.forecast_zone or not self.county_zone:
+        if not self.forecast_zone or not self.county_zone or not self.fire_weather_zone:
             await self.get_points()
+        if not self.forecast_zone or not self.county_zone or not self.fire_weather_zone:
+            raise NwsError("Forecast, county, or fire weather zone is missing")
         if not self._all_zones:
-            self._all_zones = {
+            zones_set = {
```

Comment:
What is your thought on the logic change here?  This is also potentially a breaking change.

Reply:
This was changed to make mypy happy due to a `Set[str]` being assigned to a `List[str]` variable. I couldn't get it done in a single step, but maybe there's a better way.